app/src/main/java/com/akshay/newsapp/repo/NewsRepository.kt

```diff
@@ -9,17 +9,15 @@ import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.catch
 import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.flow.flowOn
-import javax.inject.Inject
-import javax.inject.Singleton
+import retrofit2.Retrofit
 
 /**
  * Repository abstracts the logic of fetching the data and persisting it for
  * offline. They are the data source as the single source of truth.
  */
-@Singleton
-class NewsRepository @Inject constructor(
+class NewsRepository constructor(
         private val newsDao: NewsArticlesDao,
-        private val newsService: NewsService
+        private val retrofit: Retrofit
```

Comment:
I think it's better to inject the necessary service instead of whole `retrofit` instance.

Reply:
Okay.
I did this to have only one instance in the module section. As the project grows, there might be a case where we might need to add multiple service instances inside the appModule file. 

By creating the instance of Retrofit once, we have the liberty of defining the service at the point where it is used. 
Let me know what you think.

Reply:
Makes sense. But there are few issues:
- the caller of the functions of the `Repository` will be unaware of the hidden cost of creating an service instance
- each call to the function will create an instance of service
- Harder to test
- Harder to modularize

Reply:
That's true.
I'll make the necessary changes.

Reply:
Thanks 👍

Reply:
Done with this one 👍 Marking it as resolved as of now. Will push all the changes together later.